src/containers/admin-dashBoard/CoordinatorModal.tsx

```diff
@@ -34,20 +34,30 @@ export default function CoordinatorsPage() {
   });
   const [coordinators, setCoordinators] = useState<Coordinator[]>([]);
 
+  const organizationName = localStorage.getItem('orgName') as string;
   useEffect(() => {
     if (data) {
-      const extractedCoordinators = data.getAllCoordinators.map(
+      // remove organizations that the admin doesn't belongs to;
+      const newData = data.getAllCoordinators.filter((coordinator:any)=>{
+        let neededAdminOrganisations:any = null;
+        coordinator.organizations.forEach((singleOrganization: string)=> {
+          if (singleOrganization === organizationName) {
+            neededAdminOrganisations = coordinator;
+          }
+        });
+        return neededAdminOrganisations;
+      });
```

Comment:
Does this mean you get coordinators from all organizations then filter out the ones you need here? If so, why don't you filter in the backend and only return coordinators that belong to the current org? That would mean sending `organizationName` from localstorage as part of the payload.